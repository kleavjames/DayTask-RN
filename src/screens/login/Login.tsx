import React, {FC, useCallback, useState} from 'react';
import {
  Image,
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  TouchableWithoutFeedback,
  Keyboard,
  View,
  KeyboardAvoidingView,
} from 'react-native';
import auth from '@react-native-firebase/auth';
import {Formik, FormikValues} from 'formik';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import * as Yup from 'yup';
import {colors, globalStyles} from '../../themes';
import {Alert, Button, Divider, Input, Loader, Text} from '../../components';
import {
  EyelashIcon,
  GoogleIcon,
  LockIcon,
  UserTagIcon,
} from '../../assets/icons';
import {AuthStackParamList} from '../../navigation/auth';

type Props = NativeStackScreenProps<AuthStackParamList, 'Login'>;

const validationSchema = Yup.object().shape({
  email: Yup.string()
    .email('Please enter a valid email')
    .required('Email address is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
});

const Login: FC<Props> = ({navigation}) => {
  const [onLogin, setOnLogin] = useState(false);
  const [loginFailed, setOnLoginFailed] = useState(false);

  const renderGoogleIcon = () => {
    return <GoogleIcon />;
  };

  const onNavigateSignUp = () => {
    navigation.navigate('SignUp');
  };

  const onHandleSubmit = useCallback(async (values: FormikValues) => {
    try {
      setOnLogin(true);
      setOnLoginFailed(false);
      await auth().signInWithEmailAndPassword(values.email, values.password);
      setOnLogin(false);
    } catch (error) {
      console.log(error);
      setOnLogin(false);
      setOnLoginFailed(true);
    }
  }, []);

  return (
    <>
      <Loader visible={onLogin} loaderText="Logging in..." />
      <Alert
        visible={loginFailed}
        onDismiss={() => setOnLoginFailed(false)}
        dismissText="Ok">
        <Text style={globalStyles.textWhite}>Failed to login</Text>
      </Alert>
      <KeyboardAvoidingView behavior="padding" style={globalStyles.container}>
        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
          <SafeAreaView style={globalStyles.flex}>
            <View style={globalStyles.flex}>
              <Image
                style={styles.logo}
                source={require('../../assets/images/logo.png')}
                resizeMode="center"
              />
              <Text variant="titleLarge" style={styles.welcomeBack}>
                Welcome Back!
              </Text>
              <Formik
                initialValues={{email: '', password: ''}}
                onSubmit={onHandleSubmit}
                validationSchema={validationSchema}>
                {({
                  values,
                  handleChange,
                  handleSubmit,
                  handleBlur,
                  errors,
                  touched,
                }) => (
                  <>
                    <Input
                      inputLabel="Email Address"
                      testID="login-email"
                      onChangeText={handleChange('email')}
                      onBlur={handleBlur('email')}
                      value={values.email}
                      autoCapitalize="none"
                      left={<UserTagIcon />}
                      renderError={!!errors.email && !!touched.email}
                      errorMessage={errors.email}
                    />
                    <Input
                      inputLabel="Password"
                      testID="login-password"
                      secureTextEntry
                      left={<LockIcon />}
                      onChangeText={handleChange('password')}
                      onBlur={handleBlur('password')}
                      value={values.password}
                      autoCapitalize="none"
                      right={<EyelashIcon />}
                      renderRight
                      renderError={!!errors.password && !!touched.password}
                      errorMessage={errors.password}
                    />
                    <TouchableOpacity style={styles.forgotPassWrapper}>
                      <Text style={styles.forgotPass} variant="bodyMedium">
                        Forgot Password?
                      </Text>
                    </TouchableOpacity>
                    <View style={styles.loginWrapper}>
                      <Button
                        testID="login-btn"
                        mode="contained"
                        onPress={() => handleSubmit()}>
                        Log In
                      </Button>
                    </View>
                  </>
                )}
              </Formik>
              <Divider dividerText="or continue with" style={styles.divider} />
              <View>
                <Button
                  mode="outlined"
                  onPress={() => {}}
                  icon={renderGoogleIcon}>
                  Google
                </Button>
              </View>
              <Text variant="bodyMedium" style={styles.noAccount}>
                Don't have an account?{' '}
                <TouchableOpacity
                  onPress={onNavigateSignUp}
                  testID="signup-btn">
                  <Text style={globalStyles.textPrimary}>Sign Up</Text>
                </TouchableOpacity>
              </Text>
            </View>
          </SafeAreaView>
        </TouchableWithoutFeedback>
      </KeyboardAvoidingView>
    </>
  );
};

const styles = StyleSheet.create({
  logo: {
    ...globalStyles.logo,
    alignSelf: 'center',
  },
  welcomeBack: {
    color: colors.white,
    marginTop: 20,
  },
  forgotPass: {
    textAlign: 'right',
    color: colors.label,
    marginTop: 5,
  },
  forgotPassWrapper: {
    alignSelf: 'flex-end',
  },
  loginWrapper: {
    marginTop: 40,
  },
  divider: {
    marginVertical: 40,
  },
  noAccount: {
    color: colors.label,
    marginTop: 20,
    textAlign: 'center',
  },
});

export default Login;
