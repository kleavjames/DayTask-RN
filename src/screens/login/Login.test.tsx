import React from 'react';
import {render, screen, fireEvent, act} from '@testing-library/react-native';
import Login from './Login';
import {PaperProvider} from 'react-native-paper';

jest.useFakeTimers();

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe('LoginScreen', () => {
  let props: any;

  it('should render correctly', () => {
    props = createTestProps({});
    const component = (
      <PaperProvider>
        <Login {...props} />
      </PaperProvider>
    );

    const tree = render(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show the correct input validation', async () => {
    props = createTestProps({});
    const component = (
      <PaperProvider>
        <Login {...props} />
      </PaperProvider>
    );

    render(component);

    const emailAdd = screen.getByTestId('login-email');
    const password = screen.getByTestId('login-password');
    const loginBtn = screen.getByTestId('login-btn');

    await act(() => {
      fireEvent.press(loginBtn);
    });

    expect(emailAdd.props.errorMessage).toBe('Email address is required');
    expect(password.props.errorMessage).toBe('Password is required');
  });

  it('should navigate to sign up screen', async () => {
    props = createTestProps({});
    const component = (
      <PaperProvider>
        <Login {...props} />
      </PaperProvider>
    );

    render(component);

    const signup = screen.getByTestId('signup-btn');
    await act(() => {
      fireEvent.press(signup);
    });

    expect(props.navigation.navigate).toHaveBeenCalledWith('SignUp');
  });
});