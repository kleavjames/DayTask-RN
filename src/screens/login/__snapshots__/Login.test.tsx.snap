// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#212832",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Image
        resizeMode="center"
        source={
          {
            "testUri": "../../../src/assets/images/logo.png",
          }
        }
        style={
          {
            "alignSelf": "center",
            "height": 100,
            "width": 100,
          }
        }
      />
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 22,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 28,
              },
              {
                "color": "#FFFFFF",
                "marginTop": 20,
              },
            ],
          ]
        }
      >
        Welcome Back!
      </Text>
      <View
        style={
          {
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "color": "#8CAAB9",
                  "marginBottom": 10,
                },
              ],
            ]
          }
        >
          Email Address
        </Text>
        <View
          style={
            {
              "position": "relative",
            }
          }
        >
          <View
            style={
              {
                "left": 15,
                "position": "absolute",
                "top": 12,
                "zIndex": 1000,
              }
            }
          >
            <SvgMock />
          </View>
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "top": 0,
                  },
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "#455A64",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  {
                    "backgroundColor": "#455A64",
                    "borderRadius": 0,
                    "borderWidth": 1,
                  },
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 48,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  cursorColor="rgba(103, 80, 164, 1)"
                  editable={true}
                  inputLabel="Email Address"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="rgba(73, 69, 79, 1)"
                  renderError={false}
                  selectionColor="rgba(103, 80, 164, 1)"
                  style={
                    [
                      {
                        "margin": 0,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": 19.2,
                        "minWidth": 65,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      {
                        "marginLeft": 56,
                        "paddingLeft": 0,
                      },
                      {
                        "textAlign": "auto",
                      },
                    ]
                  }
                  testID="login-email"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "marginVertical": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "color": "#8CAAB9",
                  "marginBottom": 10,
                },
              ],
            ]
          }
        >
          Password
        </Text>
        <View
          style={
            {
              "position": "relative",
            }
          }
        >
          <View
            style={
              {
                "left": 15,
                "position": "absolute",
                "top": 12,
                "zIndex": 1000,
              }
            }
          >
            <SvgMock />
          </View>
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "top": 0,
                  },
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "#455A64",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  {
                    "backgroundColor": "#455A64",
                    "borderRadius": 0,
                    "borderWidth": 1,
                  },
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 48,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  cursorColor="rgba(103, 80, 164, 1)"
                  editable={true}
                  inputLabel="Password"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholderTextColor="rgba(73, 69, 79, 1)"
                  renderError={false}
                  renderRight={true}
                  secureTextEntry={true}
                  selectionColor="rgba(103, 80, 164, 1)"
                  style={
                    [
                      {
                        "margin": 0,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": 19.2,
                        "minWidth": 65,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      {
                        "marginLeft": 56,
                        "marginRight": 56,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                      },
                      {
                        "textAlign": "auto",
                      },
                    ]
                  }
                  testID="login-password"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              {
                "position": "absolute",
                "right": 15,
                "top": 12,
                "zIndex": 1000,
              }
            }
          >
            <SvgMock />
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "color": "#8CAAB9",
                  "marginTop": 5,
                  "textAlign": "right",
                },
              ],
            ]
          }
        >
          Forgot Password?
        </Text>
      </View>
      <View
        style={
          {
            "marginTop": 40,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(103, 80, 164, 1)",
              "borderRadius": 0,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="login-btn-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(103, 80, 164, 1)",
                "borderColor": "transparent",
                "borderRadius": 0,
                "borderStyle": "solid",
                "borderWidth": 0,
                "flex": undefined,
                "minWidth": 64,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="login-btn-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 0,
                  },
                ]
              }
              testID="login-btn"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "paddingVertical": 5,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                  testID="login-btn-text"
                >
                  Log In
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
            {
              "marginVertical": 40,
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgba(202, 196, 208, 1)",
                "height": 0.5,
              },
              undefined,
              false,
              false,
              {
                "borderColor": "#8CAAB9",
                "flex": 1,
              },
            ]
          }
        />
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "color": "#8CAAB9",
                  "paddingHorizontal": 20,
                },
              ],
            ]
          }
        >
          or continue with
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "rgba(202, 196, 208, 1)",
                "height": 0.5,
              },
              undefined,
              false,
              false,
              {
                "borderColor": "#8CAAB9",
                "flex": 1,
              },
            ]
          }
        />
      </View>
      <View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 0,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 0,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "minWidth": 64,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="button-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 0,
                  },
                ]
              }
              testID="button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "paddingVertical": 5,
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "marginLeft": 12,
                        "marginRight": -4,
                      },
                      {
                        "marginLeft": 16,
                        "marginRight": -16,
                      },
                      false,
                    ]
                  }
                  testID="button-icon-container"
                >
                  <SvgMock />
                </View>
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          undefined,
                          false,
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                  testID="button-text"
                >
                  Google
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              {
                "color": "#8CAAB9",
                "marginTop": 20,
                "textAlign": "center",
              },
            ],
          ]
        }
      >
        Don't have an account?
         
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="signup-btn"
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "Inter-Regular",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "#FFD56B",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
